File: content/learn/arrays-easy/06-move-zeros-to-end/solution.py
def move_zeros_to_end(a):
    write = 0
    for read in range(len(a)):
        if a[read] != 0:
            a[write], a[read] = a[read], a[write]
            write += 1


n = int(input())
a = list(map(int, input().split())) if n > 0 else []
move_zeros_to_end(a)
print(" ".join(str(x) for x in a))

File: content/learn/arrays-easy/06-move-zeros-to-end/solution.js
function moveZerosToEnd(a) {
  let write = 0;
  for (let read = 0; read < a.length; read++) {
    if (a[read] !== 0) {
      const tmp = a[write];
      a[write] = a[read];
      a[read] = tmp;
      write++;
    }
  }
}

const n = parseInt(readline(), 10);
const a = n > 0 ? readline().split(" ").map(Number) : [];
moveZerosToEnd(a);
print(a.join(" "));
